Shopify/shopify-cli
# Prevent hidden subcommands from appearing in the base command's help

### WHY are these changes introduced?

If a command is a `hidden_feature`, then it is intended to be hidden when the help is displayed. One example is the `script javy` [sub command](https://github.com/Shopify/shopify-cli/blob/78377851979e30c037389c20eaf40aa46a406831/lib/project_types/script/commands/javy.rb#L8). It is supposed to be a hidden feature. Right now, the command appears in the help menu.

<img width="829" alt="jacobsteves@Jacobs-MacBook-Pro:~:src:github" src="https://user-images.githubusercontent.com/28009669/143085456-49d7f7dc-fa5c-4ba2-a9e4-404cfc2d0faf.png">

### WHAT is this pull request doing?

Fixes the issue so only visible subcommands are shown in a command's help. 

<img width="826" alt="Screen Shot 2021-11-23 at 1 51 53 PM" src="https://user-images.githubusercontent.com/28009669/143086031-b94842de-95d6-440c-8fc8-550d1995f707.png">

<img width="884" alt="jacobsteves@Jacobs-MacBook-Pro:~:src:github" src="https://user-images.githubusercontent.com/28009669/143085944-078009a7-03b9-4c20-9a68-1a31e49bdd59.png">

### How to test your changes?

- Run `shopify script help` and `shopify help`
- Check that `javy` is not displayed as a command.

### Update checklist
